src/main/resources/native/android/dalvik/source/BleService.java

```diff
@@ -0,0 +1,214 @@
+System.err.println("BLESERVICE: ONSCANRESULT, callbacktype = "+callbackType);
+                ScanRecord mScanRecord = result.getScanRecord();
+                byte[] scanRecord = mScanRecord.getBytes();
+
+                // https://www.pubnub.com/blog/2015-04-14-building-android-beacon-android-ibeacon-tutorial-overview/
+                int index = 0;
+                while (index < scanRecord.length) {
+                    int length = scanRecord[index++];
+                    if (length == 0) {
+                        break;
+                    }
+
+                    int type = (scanRecord[index] & 0xff);
+System.err.println("BLEService, type = "+type);
+
+                    // process data
+                    if (type == 0xff) {
+                        processAD(scanRecord, index + 1, result.getRssi());
+                        // ScanDetection detection = processAD(scanRecord, index + 1, result.getRssi());
+                        // if (detection != null) {
+                            // Platform.runLater(() -> callback.accept(detection));
+                        // }
+                    }
+
+                    index += length;
+                }
+            }
+        };
+        return answer;
+    }
+
+   private void processAD(byte[] scanRecord, int init, int mRssi) {
+        // AD: (Length FF) mID0 mID1 bID1 bID0 uuid15 ... uuid0 M1 M0 m1 m0 tx
+        int startByte = init;
+        // Manufacturer ID (little endian)
+        // https://www.bluetooth.org/en-us/specification/assigned-numbers/company-identifiers
+        int mID = ((scanRecord[startByte+1] & 0xff) << 8) | (scanRecord[startByte] & 0xff);
+
+        startByte += 2;
+        // Beacon ID (big endian)
+        int beaconID = ((scanRecord[startByte] & 0xff) << 8) | (scanRecord[startByte+1] & 0xff);
+        startByte += 2;
+        // UUID (big endian)
+        byte[] uuidBytes = Arrays.copyOfRange(scanRecord, startByte, startByte+16);
+        String scannedUuid = ByteArrayToUUIDString(uuidBytes);
+System.err.println("BLEService, uuid = "+scannedUuid);
+
+        if (uuids.isEmpty() || uuids.contains(scannedUuid.toLowerCase())) {
+            startByte += 16;
+            // major (big endian)
+            int major = ((scanRecord[startByte] & 0xff) << 8) | (scanRecord[startByte+1] & 0xff);
+            startByte += 2;
+            // minor (big endian)
+            int minor = ((scanRecord[startByte] & 0xff) << 8) | (scanRecord[startByte+1] & 0xff);
+            startByte += 2;
+            // power in dB
+            int power = (scanRecord[startByte] & 0xff);
+            power -= 256;
+            double acc = calculateAccuracy(power, mRssi);
+
+            System.out.println("Scan: mID: "+mID+", beaconID: "+beaconID+", uuid: "+scannedUuid+
+                    ", major: "+major+", minor: "+minor+", power: "+power+", distance: "+acc);
+
+        }
+    }
+
+    private static String ByteArrayToUUIDString(byte[] ba) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The static method name 'ByteArrayToUUIDString' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/manual/gluonhq/substrate/pullRequest?prid=5148382)